extensions/jdbc/jdbc-mariadb/runtime/src/main/java/io/quarkus/jdbc/mariadb/runtime/graal/SimpleSocketHandlerFunction.java

```diff
@@ -0,0 +1,15 @@
+package io.quarkus.jdbc.mariadb.runtime.graal;
+
+import java.io.IOException;
+import java.net.Socket;
+
+import org.mariadb.jdbc.internal.io.socket.SocketHandlerFunction;
+import org.mariadb.jdbc.internal.util.Utils;
+import org.mariadb.jdbc.util.Options;
+
+public class SimpleSocketHandlerFunction implements SocketHandlerFunction {
+    @Override
+    public Socket apply(Options options, String host) throws IOException {
+        return Utils.standardSocket(options, host);
```

Comment:
should this perhaps log a warning or throw an exception if a MariaDB feature is being disabled?

We shouldn't silentily ignore configuration.

Reply:
Good idea yeah! Throwing an exception I think it the proper way to go. WDYT?

Reply:
Updates